app/controllers/queue_items_controller.rb

```diff
@@ -0,0 +1,38 @@
+class QueueItemsController < ApplicationController
+  before_action :require_user
+  def index
+    @queue_items = current_user.queue_items
+  end
+  
+  def create
+    video = Video.find(params[:video_id])
+    if !video_already_in_my_queue?(video)
+      QueueItem.create(user: current_user, video: video, position: new_queue_item_position)
+      flash[:notice] = "The video is added to your queue!"
+    else
+      flash[:error] = "The video is already in your queue."
+    end
+    redirect_to my_queue_path
+  end
+  
+  def destroy
+    queue_item = QueueItem.find(params[:id])
+    queue_item.destroy if queue_item.user == current_user
+    position = 1
```

Comment:
Remove this as `.each_with_index` will do the job. and the `position += 1` below too.